specification/elasticsan/resource-manager/Microsoft.ElasticSan/stable/2023-01-01/elasticsan.json

```diff
@@ -1302,6 +1316,244 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ElasticSan/elasticSans/{elasticSanName}/volumegroups/{volumeGroupName}/snapshots": {
+      "get": {
+        "tags": [
+          "Snapshots"
+        ],
+        "description": "List Snapshots in a VolumeGroup or List Snapshots by Volume (name) in a VolumeGroup using filter",
+        "operationId": "Snapshots_ListByVolumeGroup",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ElasticSanNameParameter"
+          },
+          {
+            "$ref": "#/parameters/VolumeGroupNameParameter"
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "description": "Specify $filter='volumeName eq <volume name>' to filter on volume.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Operation completed successfully; retrieved snapshots in a volume group.",
+            "schema": {
+              "$ref": "#/definitions/SnapshotList"
+            }
+          },
+          "default": {
+            "description": "Error from the ElasticSan Resource Provider",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Snapshots_ListByVolumeGroup_MaximumSet_Gen": {
+            "$ref": "./examples/Snapshots_ListByVolumeGroup_MaximumSet_Gen.json"
+          },
+          "Snapshots_ListByVolumeGroup_MinimumSet_Gen": {
+            "$ref": "./examples/Snapshots_ListByVolumeGroup_MinimumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ElasticSan/elasticSans/{elasticSanName}/volumegroups/{volumeGroupName}/snapshots/{snapshotName}": {
+      "put": {
+        "tags": [
+          "Snapshots"
+        ],
+        "description": "Create a Volume Snapshot.",
```

Comment:
Per my understanding, snapshot target to a volume. 
To align with other API, the volume name should be in the Uri Path, instead of body.

Reply:
It is how snapshots are designed.

Reply:
Could you share the detail reason for this design? 
To make the API design align with other API, can make user easy to learn how to use the API, and make the client tool/sdk based on rest API also easy for customer to learn/use. So we would like to know the business justification for this design.

Reply:
we internally thought about this topic and went with snapshots under volume groups because 

- volume group is the management entity for most of the cases and if snapshots stay directly under it, it would help users to enumerate and view the size consumed by snapshots. 

- In future if we decide to have multi volume snapshot, this arrangement would be more convenient. 

- Recommendation from ARM is to not have more than 3 levels of nesting (elasticSan/volumegroup/volume)

Reply:
For the 3 reasons:
2. For Mutiple snapshots,  it looks under volumn will make user easy to manage them, like a blob can has multiple snapshots and can easy to delete all of them from single blob.
3. Not sure if this is only recommendation or required. If only recommendation, it looks will make the API more complex and not so easy to use for customer.
- customer need query the resource ID of volumn to create snapshot, this will increase unnecessary request count , and the TPS on SRP server is limited (800 per 5 minutes for 1 region, see https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/azure-subscription-service-limits#azure-storage-resource-provider-limits)
- Customer need to prepare query string to list snapshot from volumn, which is very inconvinience.
- The usage is not aligned with other API, so not easy for customer to learn/use

Besides that, the description of this API is "Create a Volume Snapshot.", which looks not aligned.